### PR TITLE
#309 Add support to cactoos-matchers

### PR DESCRIPTION
**Pull request by GnusinPavel**
`assertThat` was replaced to `Assertion`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#353](https://github.com/yegor256/jpeek/issues/353) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#353](https://github.com/yegor256/jpeek/issues/353) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/1ceb85e1-563f-406c-ae96-c2d4281d3a7b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/jpeek/AppTest.java [hunk 1]
```diff
@@ -56,19 +57,21 @@ public void createsXmlReports() throws IOException {
...
         final Path input = Paths.get(".");
         new App(input, output).analyze();
-        MatcherAssert.assertThat(
-            Files.exists(output.resolve("LCOM.xml")),
-            Matchers.equalTo(true)
-        );
-        MatcherAssert.assertThat(
-            XSLDocument
+        new Assertion<>(
+            "LCOM.xml file exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/AppTest.java [hunk 2]
```diff
@@ -56,19 +57,21 @@ public void createsXmlReports() throws IOException {
...
+            () -> Files.exists(output.resolve("LCOM.xml")),
+            new IsTrue()
+        ).affirm();
+        new Assertion<>(
+            "Has body",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel same above and what's is "has body"? Please, use a more meaningful error message

File: src/test/java/org/jpeek/AppTest.java [hunk 3]
```diff
@@ -78,34 +81,37 @@ public void canIncludePrivateMethods() throws IOException {
         final Map<String, Object> args = new HashMap<>();
         args.put("include-private-methods", 1);
         new App(input, output, args).analyze();
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "There is no public method",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/AppTest.java [hunk 4]
```diff
@@ -78,34 +81,37 @@ public void canIncludePrivateMethods() throws IOException {
...
     public void createsIndexHtml() throws IOException {
         final Path output = Files.createTempDirectory("").resolve("x2");
         final Path input = Paths.get(".");
         new App(input, output).analyze();
-        MatcherAssert.assertThat(
-            Files.exists(output.resolve("index.html")),
-            Matchers.equalTo(true)
-        );
+        new Assertion<>(
+            "index.html file exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/AppTest.java [hunk 5]
```diff
@@ -78,34 +81,37 @@ public void canIncludePrivateMethods() throws IOException {
...
 
     @Test
     public void createsIndexXml() throws IOException {
         final Path output = Files.createTempDirectory("").resolve("x7");
         final Path input = Paths.get(".");
         new App(input, output).analyze();
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Has some metrics",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/AppTest.java [hunk 6]
```diff
@@ -114,31 +120,32 @@ public void createsIndexXml() throws IOException {
...
         final List<XML> elements = new XMLDocument(
             AppTest.class.getResourceAsStream("xsd/metric.xsd")
         ).nodes("//node()[@name]");
         final IsNot<? super List<?>> populated = new IsNot<>(
             new IsEmptyIterable<>()
         );
-        MatcherAssert.assertThat(
-            elements,
+        new Assertion<>(
+            "There are elements",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/AppTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/DefaultBaseTest.java [hunk 7]
```diff
@@ -52,10 +52,11 @@ public void listsFiles() throws IOException {
             temp.resolve("a/z.class"), "".getBytes(),
             StandardOpenOption.CREATE_NEW
         );
-        MatcherAssert.assertThat(
-            new DefaultBase(temp).files(),
+        new Assertion<>(
+            "There are more the 2 files",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/DefaultBaseTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/IndexTest.java [hunk 8]
```diff
@@ -56,25 +56,26 @@ public void setUp() throws IOException {
 
     @Test
     public void createsIndexXml() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Index xml created",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/IndexTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/MainTest.java [hunk 9]
```diff
@@ -49,9 +48,10 @@ public void createsXmlReports() throws IOException {
         final Path output = Files.createTempDirectory("").resolve("x3");
         final Path input = Paths.get(".");
         Main.main("--sources", input.toString(), "--target", output.toString());
-        MatcherAssert.assertThat(
-            Files.exists(output.resolve("LCOM5.xml")),
-            Matchers.equalTo(true)
+        new Assertion<>(
+            "Report created",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/MainTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/MainTest.java [hunk 10]
```diff
@@ -121,9 +121,10 @@ public void createsXmlReportsIfOverwriteAndTargetExists()
             "--target", target.toString(),
             "--overwrite"
         );
-        MatcherAssert.assertThat(
-            Files.exists(target.resolve("LCOM5.xml")),
-            Matchers.equalTo(true)
-        );
+        new Assertion<>(
+            "Report created if target exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/MainTest.java]: @GnusinPavel please, change it to start with "Must..."

File: src/test/java/org/jpeek/MatrixTest.java [hunk 11]
```diff
@@ -56,25 +56,26 @@ public void setUp() throws IOException {
 
     @Test
     public void createsMatrixXml() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Matrix created",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/MatrixTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/MetricsTest.java [hunk 12]
```diff
@@ -251,8 +251,11 @@ public void testsTarget() throws IOException {
         } else {
             xpath = "//class[@id='%s' and number(@value)=%.4f]";
         }
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            new FormattedText(
+                "Target '%s', value '%s'", this.target, this.value
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/MetricsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportTest.java [hunk 13]
```diff
@@ -51,14 +51,16 @@
...
-        MatcherAssert.assertThat(
-            Files.exists(output.resolve("LCOM.xml")),
-            Matchers.equalTo(true)
-        );
-        MatcherAssert.assertThat(
-            Files.exists(output.resolve("LCOM.html")),
-            Matchers.equalTo(true)
-        );
+        new Assertion<>(
+            "LCOM.xml file exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportTest.java [hunk 14]
```diff
@@ -51,14 +51,16 @@
...
+            () -> Files.exists(output.resolve("LCOM.xml")),
+            new IsTrue()
+        ).affirm();
+        new Assertion<>(
+            "LCOM.html file exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportTest.java [hunk 15]
```diff
@@ -73,30 +75,32 @@ public void createsXmlReportWithXpaths() throws IOException {
             ).xml(),
             "LCOM"
         ).save(output);
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "LCOM report created",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportTest.java [hunk 16]
```diff
@@ -119,8 +123,9 @@ public void createsFullXmlReport() throws IOException {
             ),
             "LCOM"
         ).save(output);
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Full report created",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportTest.java [hunk 17]
```diff
@@ -132,21 +137,22 @@ public void createsFullXmlReport() throws IOException {
...
 
     @Test
-    public void setsCorrectschemaLocation() throws IOException {
+    public void setsCorrectSchemaLocation() throws IOException {
         final Path output = Files.createTempDirectory("");
         new Report(new Skeleton(new FakeBase()).xml(), "LCOM").save(output);
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Schema location",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/ReportWithStatisticsTest.java [hunk 18]
```diff
@@ -60,12 +60,13 @@ public void createsXml() {
         }
         latch.countDown();
         service.shutdown();
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Report with statics",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/ReportWithStatisticsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 19]
```diff
@@ -46,8 +45,9 @@
 
     @Test
     public void createsXml() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Overload Bar methods",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 20]
```diff
@@ -64,13 +64,14 @@ public void createsXml() {
...
+        ).affirm();
     }
 
     @Test
-    public void findsMethodsAndArgs() throws IOException {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+    public void findsMethodsAndArgs() {
+        new Assertion<>(
+            "Methods with diff param types",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 21]
```diff
@@ -84,13 +85,14 @@ public void findsMethodsAndArgs() throws IOException {
...
+        ).affirm();
     }
 
     @Test
-    public void findsMethodCalls() throws IOException {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+    public void findsMethodCalls() {
+        new Assertion<>(
+            "Method calls",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 22]
```diff
@@ -104,13 +106,14 @@ public void findsMethodCalls() throws IOException {
...
-        );
+        ).affirm();
     }
 
     @Test
     public void createsOnlyOneMethodIgnoresSynthetic() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "One method creates lambda",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 23]
```diff
@@ -119,13 +122,14 @@ public void createsOnlyOneMethodIgnoresSynthetic() {
...
-        );
+        ).affirm();
     }
 
     @Test
     public void findFieldWithQualifiedName() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Field with qualified name",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 24]
```diff
@@ -139,13 +143,14 @@ public void findFieldWithQualifiedName() {
...
-        );
+        ).affirm();
     }
 
     @Test
     public void findSchemaOfSkeleton() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Different method visibilities",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/SkeletonTest.java [hunk 25]
```diff
@@ -155,13 +160,14 @@ public void findSchemaOfSkeleton() {
...
-        );
+        ).affirm();
     }
 
     @Test
     public void recognizesPublicMethods() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Recognizes public methods",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/SkeletonTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/TypesOfTest.java [hunk 26]
```diff
@@ -40,8 +40,9 @@
 
     @Test
     public void parsesSignature() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Parse signature",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/TypesOfTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java [hunk 27]
```diff
@@ -41,19 +41,21 @@
 
     @Test
     public void parsesClass() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(this.classAsXml("Bar")),
+        new Assertion<>(
+            "Parse class",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/XmlClassTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/skeleton/XmlClassTest.java [hunk 28]
```diff
@@ -41,19 +41,21 @@
...
-        );
+        ).affirm();
     }
 
     @Test
     public void parsesMethodVisibility() {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Parse method visibility",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/skeleton/XmlClassTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/AsyncReportsTest.java [hunk 29]
```diff
@@ -63,14 +63,16 @@ public void rendersOneReport() throws Exception {
             );
         final Response response =
             bifunc.apply("org.jpeek", "jpeek").apply("index.html");
-        MatcherAssert.assertThat(
-            response,
+        new Assertion<>(
+            "Has HTTP OK response",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/AsyncReportsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/AsyncReportsTest.java [hunk 30]
```diff
@@ -63,14 +63,16 @@ public void rendersOneReport() throws Exception {
...
+            () -> response,
             new HmRsStatus(HttpURLConnection.HTTP_OK)
-        );
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(new RsPrint(response).printBody()),
+        ).affirm();
+        new Assertion<>(
+            "Has body",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/AsyncReportsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/MistakesTest.java [hunk 31]
```diff
@@ -53,16 +52,17 @@ public void acceptsAndRenders() throws Exception {
         new App(input, output).analyze();
         final Mistakes mistakes = new Mistakes();
         mistakes.add(output);
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Avg",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/MistakesTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/ReportsTest.java [hunk 32]
```diff
@@ -55,13 +55,13 @@ public void weAreOnline() {
 
     @Test
     @Ignore
-    public void rendersOneReport() throws Exception {
-        MatcherAssert.assertThat(
-            new Reports(
+    public void rendersOneReport() {
+        new Assertion<>(
+            "HTTP OK response",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/ReportsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/ResultsTest.java [hunk 33]
```diff
@@ -53,16 +52,17 @@ public void acceptsAndRenders() throws Exception {
         new App(input, output).analyze();
         final Results results = new Results();
         results.add("org.takes:takes", output);
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+        new Assertion<>(
+            "Repos tag exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/ResultsTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/TkIndexTest.java [hunk 34]
```diff
@@ -40,13 +39,14 @@
 public final class TkIndexTest {
 
     @Test
-    public void rendersIndexPage() throws IOException {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+    public void rendersIndexPage() {
+        new Assertion<>(
+            "Footer exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/TkIndexTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/TkMistakesTest.java [hunk 35]
```diff
@@ -41,19 +40,21 @@
 public final class TkMistakesTest {
 
     @Test
-    public void rendersMistakesPage() throws IOException {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+    public void rendersMistakesPage() {
+        new Assertion<>(
+            "Footer exists on mistakes page",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/TkMistakesTest.java]: @GnusinPavel same above.

File: src/test/java/org/jpeek/web/TkMistakesTest.java [hunk 36]
```diff
@@ -41,19 +40,21 @@
...
+        ).affirm();
     }
 
     @Test
-    public void rendersMistakesPageInXml() throws IOException {
-        MatcherAssert.assertThat(
-            XhtmlMatchers.xhtml(
+    public void rendersMistakesPageInXml() {
+        new Assertion<>(
+            "Worst tag exists",
```
Comment (fabriciofx) [inline src/test/java/org/jpeek/web/TkMistakesTest.java]: @GnusinPavel same above.

**Comment by fabriciofx** [comment]
@GnusinPavel ping

**Comment by fabriciofx** [comment]
@GnusinPavel it seems fine to me. @paulodamaso can you merge it?

**Comment by GnusinPavel** [comment]
@paulodamaso ping )

**Comment by fabriciofx** [comment]
@GnusinPavel why did you pushed again this PR after I approved it?

**Comment by GnusinPavel** [comment]
> @GnusinPavel why did you pushed again this PR after I approved it?

there were a conflict

**Comment by fabriciofx** [comment]
@GnusinPavel ah, ok. Thanks!

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#353](https://github.com/yegor256/jpeek/issues/353) is already in the agenda of @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
The job [#353](https://github.com/yegor256/jpeek/issues/353) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by fabriciofx** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/jpeek/issues/353#issuecomment-557802389))

@fabriciofx This is what I know about this job in [C7JGJ00DP](https://www.0crat.com/p/C7JGJ00DP), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#353](https://github.com/yegor256/jpeek/issues/353) is not in scope
  * The job is waiting quality review verdict by @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for 2days
  * @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C7JGJ00DP?q=%7Bjob%3A%27gh%3Ayegor256%2Fjpeek%23353%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by fabriciofx** [comment]
@GnusinPavel according with the [§42 policy](http://www.zerocracy.com/policy.html#42): *Messages in a job always start with a name of a user they are addressed to*. So, fix [here](https://github.com/yegor256/jpeek/pull/353#issuecomment-556300805).

**Comment by GnusinPavel** [comment]
@fabriciofx fixed

**Comment by fabriciofx** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Pull request merged**